core/src/main/java/fj/Hash.java

```diff
@@ -262,6 +262,32 @@ public int hash(final A a) {
     });
   }
 
+  /**
+   * A hash instance for the {@link Zipper} type.
+   *
+   * @param ha A hash for the elements of the zipper.
+   * @return A hash instance for the {@link Zipper} type.
+   */
+  public static <A> Hash<Zipper<A>> zipperHash(final Hash<A> ha) {
+    return hash(as -> {
+        final int p = 419;
+        int r = 239;
+        Zipper<A> aas = as;
+
+        aas.move(0);
```

Comment:
this does nothing?

Reply:
The zipper handed out might not be focus at start. Let me know if you think they are not needed.

Reply:
Same as for `Equal`, I'd say we just need to combine the hash of `lefts()`,` focus()` and `right()`.

Reply:
updated